lib/hash_ring/ets.ex

```diff
@@ -287,83 +349,184 @@ defmodule ExHashRing.HashRing.ETS do
     {:reply, {:ok, []}, state}
   end
 
-  def handle_call(:force_gc, _from, %{pending_gcs: pending_gcs, table: table} = state) do
+  def handle_call(:force_gc, _from, %__MODULE__{} = state) do
     ring_gens =
-      for {ring_gen, timer_ref} <- pending_gcs do
+      for {ring_gen, timer_ref} <- state.pending_gcs do
         Process.cancel_timer(timer_ref)
-        do_ring_gen_gc(table, ring_gen)
+        :ok = do_ring_gen_gc(state.current_table, ring_gen)
+        :ok = do_ring_gen_gc(state.previous_table, ring_gen)
         ring_gen
       end
 
-    {:reply, {:ok, ring_gens}, %{state | pending_gcs: %{}}}
+    {:reply, {:ok, ring_gens}, %__MODULE__{state | pending_gcs: %{}}}
   end
 
-  def handle_call({:force_gc, ring_gen}, _from, %{pending_gcs: pending_gcs, table: table} = state) do
+  def handle_call({:force_gc, ring_gen}, _from, %__MODULE__{} = state) do
     {reply, pending_gcs} =
-      case Map.pop(pending_gcs, ring_gen) do
+      case Map.pop(state.pending_gcs, ring_gen) do
         {nil, pending_gcs} ->
           {{:error, :not_pending}, pending_gcs}
 
         {timer_ref, pending_gcs} ->
           Process.cancel_timer(timer_ref)
-          {do_ring_gen_gc(table, ring_gen), pending_gcs}
+          :ok = do_ring_gen_gc(state.current_table, ring_gen)
+          :ok = do_ring_gen_gc(state.previous_table, ring_gen)
+          {:ok, pending_gcs}
       end
 
-    {:reply, reply, %{state | pending_gcs: pending_gcs}}
+    {:reply, reply, %__MODULE__{state | pending_gcs: pending_gcs}}
   end
 
-  def handle_info({:gc, ring_gen}, %{pending_gcs: pending_gcs, table: table} = state) do
+  def handle_info({:gc, ring_gen}, %__MODULE__{} = state) do
     pending_gcs =
-      case Map.pop(pending_gcs, ring_gen) do
+      case Map.pop(state.pending_gcs, ring_gen) do
         {nil, pending_gcs} ->
           pending_gcs
 
         {_stale_timer_ref, pending_gcs} ->
-          do_ring_gen_gc(table, ring_gen)
+          :ok = do_ring_gen_gc(state.previous_table, ring_gen)
```

Comment:
`previous_table` would only have `ring_gen -1` right?

Reply:
A "Generation" is both the "current" and "previous" configuration so the previous_table and the current_table use the same generation.